providers/src/airflow/providers/edge/worker_api/datamodels.py

```diff
@@ -53,19 +55,56 @@ class JsonRpcRequest(JsonRpcRequestBase):
     )
 
 
-class WorkerStateBody(BaseModel):
+class EdgeJobBase(BaseModel):
+    """Basic attributes of a job on the edge worker."""
+
+    dag_id: str = Field(title="Dag ID", description="Identifier of the DAG to which the task belongs.")
+    task_id: str = Field(title="Task ID", description="Task name in the DAG.")
+    run_id: str = Field(title="Run ID", description="Run ID of the DAG execution.")
+    map_index: int = Field(
+        title="Map Index",
+        description="For dynamically mapped tasks the mapping number, -1 if the task is not mapped.",
+    )
```

Comment:
While we have a comment about using `str` because of connexion, we have it as int here? Can we make it one or the other in the entire edge provider?

Reply:
Just Flask has a parsing problem and therefore the Airflow 2.10 REST implementation in Connexion uses "string" for the map-index as there can be negative values. But I convert this to int() before going into the logic.

The primary implementation is FastAPI and this can correctly treat negative integer values. So just in the openapi specs of connexion and the `_v2_routes.py` the map_index is treated as string to make it working. Otherwise FastAPI and Pydantic models should be consistent.